source/Irrlicht/CQuantNormalCache.h

```diff
@@ -8,69 +8,69 @@ namespace irr
-	void insertIntoCache8_8_8(const core::vectorSIMDf& key, const uint32_t quantizedNormal);
-	void insertIntoCache16_16_16(const core::vectorSIMDf& key, const uint32_t quantizedNormal);
+struct VectorUV
+{
+	float u;
+	float v;
 
-private:
-	struct VectorUV
+	inline bool operator==(const VectorUV& other) const
 	{
-		float u;
-		float v;
+		return (u == other.u && v == other.v);
+	}
+};
 
-		inline bool operator==(const VectorUV& other) const
-		{
-			return (u == other.u && v == other.v);
-		}
-	};
+struct Vector16u
+{
+	uint16_t x;
+	uint16_t y;
+	uint16_t z;
+};
 
-	struct Vector16u
+struct Vector8u
+{
+	uint8_t x;
+	uint8_t y;
+	uint8_t z;
+};
+
+struct QuantNormalHash
+{
+	inline size_t operator()(const VectorUV& vec) const noexcept
 	{
-		uint16_t x;
-		uint16_t y;
-		uint16_t z;
-	};
+		static constexpr size_t primeNumber1 = 18446744073709551557;
+		static constexpr size_t primeNumber2 = 4611686018427388273;
 
-	struct Vector8u
+		return  ((static_cast<size_t>(static_cast<double>(vec.u)* std::numeric_limits<size_t>::max())* primeNumber1) ^
+			(static_cast<size_t>(static_cast<double>(vec.v)* std::numeric_limits<size_t>::max())* primeNumber2));
+	}
+};
+
+struct QuantNormalEqualTo
+{
+	inline bool operator()(const core::vectorSIMDf& lval, const core::vectorSIMDf& rval) const noexcept
 	{
-		uint8_t x;
-		uint8_t y;
-		uint8_t z;
-	};
+		return (lval == rval).all();
+	}
 
-	struct QuantNormalHash
+	inline bool operator()(const VectorUV& lval, const VectorUV& rval) const noexcept
 	{
-		inline size_t operator()(const VectorUV& vec) const noexcept
-		{
-			static constexpr size_t primeNumber1 = 18446744073709551557;
-			static constexpr size_t primeNumber2 = 4611686018427388273;
+		return (lval.u == rval.u && lval.v == rval.v);
+	}
+};
 
-			return  ((static_cast<size_t>(static_cast<double>(vec.u)* std::numeric_limits<size_t>::max())* primeNumber1) ^
-				(static_cast<size_t>(static_cast<double>(vec.v)* std::numeric_limits<size_t>::max())* primeNumber2));
-		}
-	};
+class CQuantNormalCache
+{ 
+public:
+	static uint32_t quantizeNormal2_10_10_10(const core::vectorSIMDf& normal);
+	static uint32_t quantizeNormal8_8_8(const core::vectorSIMDf& normal);
+	static uint64_t quantizeNormal16_16_16(const core::vectorSIMDf& normal);
 
-	struct QuantNormalEqualTo
-	{
-		inline bool operator()(const core::vectorSIMDf& lval, const core::vectorSIMDf& rval) const noexcept
-		{
-			return (lval == rval).all();
-		}
-
-		inline bool operator()(const VectorUV& lval, const VectorUV& rval) const noexcept
-		{
-			return (lval.u == rval.u && lval.v == rval.v);
-		}
-	};
-
-	inline VectorUV mapToBarycentric(const core::vectorSIMDf& vec) const
+	static void insertIntoCache2_10_10_10(const core::vectorSIMDf& key, const uint32_t quantizedNormal);
+	static void insertIntoCache8_8_8(const core::vectorSIMDf& key, const uint32_t quantizedNormal);
+	static void insertIntoCache16_16_16(const core::vectorSIMDf& key, const uint32_t quantizedNormal);
```

Comment:
dont make the caches static, I hate globals!

Reply:
@devshgraphicsprogramming should i make this class singleton then? I don't think there is need for more than one instance of this class

Reply:
no singletons, no globals, user might want to make bazillion instances of asset manager (and ergo normal caches)